lib/spack/external/archspec/json/cpu/microarchitectures.json

```diff
@@ -2706,7 +2706,7 @@
         ],
         "clang" : [
           {
-            "versions": "9.0:12.0",
+            "versions": "9.0:12.99",
```

Comment:
Is this intentional?  I think this should be a separate change?

Reply:
I think this is related to the bug that @haampie was discussing reverting a PR for about version ranges with upper bounds. I can revert this and if it fails in CI we can look at fixing the other bug.